File: INF.04-25.01/INF.04-02-25.01-SG/konsolowa/py/main.py
class Device:
    @staticmethod
    def show_message(message: str) -> None:
        """
        ******************
        Nazwa: show_message
        Opis: metoda wyświetla w konsoli wiadomość podaną jako parametr.
        Parametry: message - wiadomość do wyświetlenia
        Zwracany typ i opis: brak
        Autor: 123456789112
        ******************
        """
        print(message)


class WashingMachine(Device):
    def __init__(self):
        self.__program = 0

    def set_program(self, program: int) -> int:
        if 1 <= program <= 12:
            self.__program = program
        else:
            self.__program = 0

        return self.__program


class VacuumCleaner(Device):
    def __init__(self):
        self.__status = False

    def on(self):
        if not self.__status:
            self.__status = True
            Device.show_message("Odkurzacz włączono")

    def off(self):
        if self.__status:
            self.__status = False
            Device.show_message("Odkurzacz wyłączono")


def main():
    vacuum = VacuumCleaner()
    washing_machine = WashingMachine()

    program = int(input("Podaj numer prania 1..12: "))
    if program == washing_machine.set_program(program):
        print("Program został ustawiony")
    else:
        print("Podano niepoprawny numer programu")

    vacuum.on()
    vacuum.on()
    vacuum.on()
    Device.show_message("Odkurzacz wyładował się")
    vacuum.off()


if __name__ == '__main__':
    main()

File: INF.04-25.01/INF.04-02-25.01-SG/konsolowa/cpp/main.cpp
#include <iostream>

class Device {
public:
    /* ******************
     * Nazwa: showMessage
     * Opis: metoda wyświetla w konsoli wiadomość podaną jako parametr.
     * Parametry: message - wiadomość do wyświetlenia
     * Zwracany typ i opis: brak
     * Autor: 123456789112
     * ******************
     */
    static void showMessage(const std::string &message) {
        std::cout << message << std::endl;
    }
};

class WashingMachine : public Device {
private:
    int washingProgram = 0;

public:
    int setWashingProgram(const int washingProgram) {
        if (washingProgram >= 1 && washingProgram <= 12) {
            this->washingProgram = washingProgram;
        } else {
            this->washingProgram = 0;
        }
        return this->washingProgram;
    }
};

class VacuumCleaner : public Device {
private:
    bool status = false;

public:
    void on() {
        if (this->status == false) {
            this->status = true;
            Device::showMessage("Odkurzacz włączono");
        }
    }

    void off() {
        if (this->status == true) {
            this->status = false;
            Device::showMessage("Odkurzacz wyłączono");
        }
    }
};

int main() {
    auto vacuum = VacuumCleaner();
    auto washingMachine = WashingMachine();

    int washingProgram;
    std::cout << "Podaj numer prania 1..12: ";
    std::cin >> washingProgram;

    if (washingProgram == washingMachine.setWashingProgram(washingProgram)) {
        std::cout << "Program został ustawiony" << std::endl;
    } else {
        std::cout << "Podano niepoprawny numer programu" << std::endl;
    }

    vacuum.on();
    vacuum.on();
    vacuum.on();
    Device::showMessage("Odkurzacz wyładował się");
    vacuum.off();

    return 0;
}
